libyul/backends/wasm/WordSizeTransform.h

```diff
@@ -0,0 +1,69 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Replace every u256 variable with four u64 variables.
+ */
+
+#pragma once
+
+#include <libyul/optimiser/ASTWalker.h>
+#include <libyul/optimiser/NameDispenser.h>
+
+#include <array>
+#include <vector>
+
+namespace yul
+{
+
+/**
+ * A stage that replace every u256 variable with four u64 variables.
+ * This transformation stage is required because values in EVM are 256 bits,
+ * but wasm only supports values up to 64 bits, so we use four u64 values to simulate
+ * one u256 value.
+ *
```

Comment:
Actually, this should have a comment saying that it accepts code with EVM FunctionalInstructions, but outputs code that uses EVM opcodes for FunctionalInstructions, albout variations that take and return four times the number of variables.

Thinking about that, it might be better to first transform the code from FunctionalInstruction to FuntionCall (we want to do that anyway in the long run), which would then also simplify the code because only one kind is needed - but not sure about the amount of simplification.

Reply:
Ok, it saves 4-5 lines, so maybe not worth it. But please add the comment.

Reply:
Done